scripts/hid_configurator/configurator_core.py

```diff
@@ -315,9 +353,77 @@ def open_device(device_type):
     return dev
 
 
-def dfu_sync(dev, args):
+def fwinfo(dev, recipient):
+    event_id = (EVENT_GROUP_DFU << GROUP_FIELD_POS) | (DFU_IMGINFO << TYPE_FIELD_POS)
+
+    try:
+        success, fetched_data = exchange_feature_report(dev, recipient, event_id, None, True)
```

Comment:
this exception checking pattern occurs in many places and could actually be kept inside `exchange_feature_report` so it returns `success=False` in such case

Reply:
Done